src/platform/Zephyr/PlatformManagerImpl.cpp

```diff
@@ -37,19 +38,29 @@ static K_THREAD_STACK_DEFINE(sChipThreadStack, CHIP_DEVICE_CONFIG_CHIP_TASK_STAC
 
 PlatformManagerImpl PlatformManagerImpl::sInstance{ sChipThreadStack };
 
+static int app_entropy_source(void * data, unsigned char * output, size_t len, size_t * olen)
+{
+    const struct device * entropy = device_get_binding(DT_CHOSEN_ZEPHYR_ENTROPY_LABEL);
+
+    SuccessOrExit(entropy_get_entropy(entropy, output, len));
+
+    *olen = len;
+
+exit:
```

Comment:
please remove this goto exit pattern - no cleanup code and it seems to make the code more complex to follow.

```cpp

if (entropy_get_entropy(entropy, output, len) == 0) {
   *olen = len;
} else {
   *olen = 0;  // is this ok? should we set any value?
}
```

`entropy_get_entropy` does not return a CHIP_ERROR so we should not use the SuccessOrExit macro which is supposed to work on CHIP_ERROR.